### ISSUE TITLE
Constructor from Generated Code?

### ISSUE THREAD

**Issue by xanderdunn** [tanmaykm/ProtoBuf.jl#53]
I have this type in my .proto:
```
message Thingie {
    required int32 value1 = 1;
    required int32 value2 = 2;
    optional bool switch = 3;
}
```

Using protoc, this is generated into:
```julia
type Thingie
    value1::Int32
    value2::Int32
    switch::Bool
    Thingie() = (o=new(); fillunset(o); o)
end #type Thingie
```

When I execute `Thingie(Int32(0), Int32(0), false)`, I get:
```
ERROR: LoadError: LoadError: MethodError: `convert` has no method matching convert(::Type{Thingie}, ::Int32, ::Int32, ::Bool)
This may have arisen from a call to the constructor Thingie(...),
since type constructors fall back to convert methods.
```

Indeed, it appears there is no constructor in the generated code that supports the three inputs?  If I simply modify the generated code so that default constructor is generated under the hood, it works:
```julia
type Thingie
    value1::Int32
    value2::Int32
    switch::Bool
end #type Thingie
```

There's a good chance I'm using it incorrectly, but this isn't what I expected?  And thanks @tanmaykm for making this!

**Comment by xanderdunn** [comment]
Hmm... it looks like from your comment on #9 that it's necessary to write my own constructors:
```julia
function Thingie(value1::Int32, value2::Int32, switch::Bool)
    thingie = Thingie()
    thingie.value1 = value1
    fillset(thingie, :value1)
    action.value2 = value2
    fillset(thingie, :value2)
    action.switch = switch
    fillset(thingie, :switch)
    return thingie
end
```

Is it possible to make an option for generating default non-empty constructors?  In all cases where I use Protobuf, the default constructors are all I'll need, so it would be convenient to have them generated automatically.  Other languages, like the Java and Python Protobuf generators, also produce default, non-empty constructors. 

A discussion in the README on constructors could help clear this up.

**Comment by tanmaykm** [comment]
Yes, it is possible to generate a non-empty constructor that initializes all members. But what is not clear to me is how best to allow for optionally leaving out some non-mandatory members. Probably generate them as keyword arguments?

**Comment by tanmaykm** [comment]
@xanderdunn do take a look at #56

**Comment by tanmaykm** [comment]
closed via #56
